ratatui-widgets/src/table.rs

```diff
@@ -902,6 +909,84 @@ impl Table<'_> {
         }
     }
 
+    /// Render cells into the columns of a row
+    ///
+    /// Render `Cell`s from `cells` into columns consumed from `column_widths_iterator`, stopping
+    /// if either of these iterators are finished.  Each `Cell` gets rendered across
+    /// [`Cell::get_column_span`] columns plus the gaps between them, if this value is > 1.
+    fn render_row_cells<'a, T>(
+        &self,
+        buf: &mut Buffer,
+        column_widths_iterator: &mut T,
```

Comment:
I think this can probably be just Vec tather than the iterator approach. I'd use an iter here if there was state that needed to be kept about where we got up to in the iterator between method calls, but it seems like this doesn't need that statefulness (unless I'm mistaken in the call pattern). This will simplify this a bit.